### ISSUE TITLE
Question: Clarification on building a reference

### ISSUE THREAD

**Issue by FriederikeHanssen** [etal/cnvkit#836]
Hi! 

Apologies if this is answered elsewhere and I didn't find it. I was wondering if you could confirm that I am using CNVKit appropriately:

I have a cohort of all-male WGS samples tumor and normal, but unpaired samples. They were aligned with BWA, duplicate marked, and base quality recalibrated. The BQSR step was run on accessible regions as provided by GATK https://gatk.broadinstitute.org/hc/en-us/articles/360035889551-When-should-I-restrict-my-analysis-to-specific-intervals- to eliminate "diffiult" regions. Reads in those regions are removed. Would you recommend adding the region file in as bed file as well?

I was planning to create a reference from the normal samples with:

```
cnvkit.py batch -n *normals.cram --method wgs --fasta reference.fasta
```
--> should output reference.cnn. Do I need to add `--male-reference` here as well

```
cnvkit.py batch *tumor.cram --method wgs --male-reference --reference reference.cnn --scatter --diagram --output-dir ./results
```

Does that seem sensible?

And in general for running CNVkit also with WES or target panel data, is it sufficient to run `cnvkit batch` by providing the target file and not precompute the target and antitarget.cnn?

So basically:

WES/Panel:
```
cnvkit.py batch tumor.cram normal.cram --method hybrid/amplicon --target my_target.bed --scatter --diagram --output-dir ./results
```
 is sufficient?

Thank you very much for your help! 
Rike

**Comment by etal** [comment]
For WGS on the post-BQSR BAM where all reads in difficult/repetitive regions have been removed, you may need to run in "amplicon" mode with the accessible regions as your "targets" in order to force CNVkit to skip those regions. Otherwise, I think "wgs" mode ignores sequencing accessibility since that was originally intended to be a filter on off-target regions that were not considered in the capture design. Or, you can run it on the pre-BQSR BAM in "wgs"; the main difference is the default bin size.

For WES or panel, yes, you can run `batch` with the capture targets BED as input and it will figure out appropriate bin sizes and sequencing accessibility. I recommend using the pre-BQSR BAM as input for these samples because CNVkit uses the copy number signal from off-target reads, and handles "difficult" genomic regions in its own way. If your pipeline has already filtered out all reads in the the off-target and difficult regions, then these post-BQSR BAMs would be better analyzed in `amplicon` mode with the intersection of capture targets and accessible regions as the input -- i.e. all the locations where there are still reads remaining after the filter.

You don't strictly need to use `--male-reference` (a.k.a. haploid X reference) to analyze male samples. The calls and exported VCF will work the same either way (assuming the option is used consistently throughout the pipeline); the main difference is that the plots will show haploid X as neutral copy number, e.g. `scatter` will plot a single copy of X with a log2 value of 0.0 instead of -1.0. It's kind of a legacy feature.

**Comment by FriederikeHanssen** [comment]
Awesome, thank you so much for the detailed answer 🙏 😃

**Comment by zhaowsong** [comment]
> For WGS on the post-BQSR BAM where all reads in difficult/repetitive regions have been removed, you may need to run in "amplicon" mode with the accessible regions as your "targets" in order to force CNVkit to skip those regions. Otherwise, I think "wgs" mode ignores sequencing accessibility since that was originally intended to be a filter on off-target regions that were not considered in the capture design. Or, you can run it on the pre-BQSR BAM in "wgs"; the main difference is the default bin size.
> 
> For WES or panel, yes, you can run `batch` with the capture targets BED as input and it will figure out appropriate bin sizes and sequencing accessibility. I recommend using the pre-BQSR BAM as input for these samples because CNVkit uses the copy number signal from off-target reads, and handles "difficult" genomic regions in its own way. If your pipeline has already filtered out all reads in the the off-target and difficult regions, then these post-BQSR BAMs would be better analyzed in `amplicon` mode with the intersection of capture targets and accessible regions as the input -- i.e. all the locations where there are still reads remaining after the filter.
> 
> You don't strictly need to use `--male-reference` (a.k.a. haploid X reference) to analyze male samples. The calls and exported VCF will work the same either way (assuming the option is used consistently throughout the pipeline); the main difference is that the plots will show haploid X as neutral copy number, e.g. `scatter` will plot a single copy of X with a log2 value of 0.0 instead of -1.0. It's kind of a legacy feature.

Hello, I'd like to call CNVs on BQSR-processed BAM files. Is my code correct?

cnvkit.py batch \
          ${input_dir}/*T_bqsr.bam \
          --normal ${input_dir}/*G_bqsr.bam \
          --method amplicon \
          --fasta ${ref}
          --annotate ${refFlat} \
          -p 8 \
          -d ${output_dir} \
          --drop-low-coverage \
          --scatter --diagram \
          --output-dir ${output_dir}
